### PR TITLE
Archive spells

### PR DESCRIPTION
**Pull request by iamchrissmith**
Make a script to archive the spell each week and organize the archive directory to have subdirectories that contain each week's archive

Tests should pass with: HardHat Fork pinned at block n. 5749312

currently dependent on https://github.com/makerdao/spells-goerli/pull/23, once that is merged, switch base to master

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -5,3 +5,4 @@ clean  :; dapp clean
 test   :; ./test-dssspell.sh $(match)
 deploy :; make && dapp create DssSpell | xargs ./verify.py DssSpell
 flatten :; hevm flatten --source-file "src/Goerli-DssSpell.sol" > out/flat.sol
+archive-spell :; ./archive-spell.sh $(name)
```
Comment (naszam) [inline Makefile:8]: I would name it just archive, to keep it short.

File: archive-spell.sh [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+if [[ -z "$1" ]]; then
+  echo "You must provide a name option to name the directory"
+else
+  mkdir "./archive/$name"
+  cp "./src/Goerli-DssSpell.sol" "./archive/$name"
+  cp "./src/Goerli-DssSpell.t.sol" "./archive/$name"
+  cp "./src/Goerli-DssSpell.t.base.sol" "./archive/$name"
+  echo "Spell, tests and base copied to archive directory"
+fi
```
Comment (naszam) [inline archive-spell.sh:13]: nit: missing new line

File: Makefile (hunk 1 above)
Comment (iamchrissmith) [reply Makefile:8]: i tried that and got `make: `archive' is up to date.` returned which made me think that `make archive` is a reserved command.

Comment (naszam) [reply Makefile:8]: Oh! right.

Comment (naszam) [inline Makefile:8]: nit: missing newline

Comment (naszam) [inline Makefile:8]: should we keep the `dssspell` naming convention in the script file?

Comment (iamchrissmith) [reply Makefile:8]: yeah i'll update the file name, good call

Comment (gbalabasquer) [reply Makefile:8]: Done in my PR after merging this.

**Pull request merged**